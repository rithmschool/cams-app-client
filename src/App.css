body {
  font-family: -apple-system, BlinkMacSystemFont, sans-serif;
  color: #4d4d4d;
  background: linear-gradient(135deg, #00d8ff 0%, #7db9e8 100%);
  background-size: cover; }

a {
  color: #00c5cc;
  text-decoration: none; }

.serif {
  font-family: Georgia, serif; }

.white {
  color: white; }

.bold-shadow {
  text-shadow: -2px 2px 0px #009eb8, -3px 3px 0px #00aea7;
  color: white; }

.center {
  text-align: center;
  margin: auto; }

.right {
  text-align: right; }

.xs {
  font-size: .4rem; }

.sm {
  font-size: .6rem; }

.me {
  font-size: 1rem; }

.lg {
  font-size: 1.5rem; }

.xl {
  font-size: 2.5rem; }

button {
  cursor: pointer; }

/* NAV
*********/
.nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
  background: rgba(255, 255, 255, 0.9);
  border-bottom: 2px solid rgba(0, 0, 0, 0.1); }

.logo {
  margin: 0.5rem;
  justify-content: flex-start;
  align-content: flex-start;
  border: 3px solid white;
  border-radius: 3px;
  box-shadow: 0px 1px 3px rgba(128, 128, 128, 0.13); }

.button {
  background: white;
  border: 3px solid white;
  border-radius: 3px;
  padding: 0.7rem;
  box-shadow: 0px 1px 3px rgba(128, 128, 128, 0.13);
  color: #00c5cc;
  margin: 0.5rem; }

.button:hover {
  text-decoration: none; }

/*BANNER
*********/
.banner {
  width: 120vw;
  z-index: -1;
  position: fixed;
  margin-top: -7rem; }

.banner-text {
  padding: 0 1rem 0 1rem;
  color: white; }

h1.banner-bold {
  text-shadow: -2px 2px 0px #009eb8, -3px 3px 0px #00aea7;
  color: white;
  font-style: italic;
  text-transform: uppercase;
  max-width: 60vw; }

/* CONTENT
*********/
.content {
  margin-top: 1rem;
  text-align: center;
  padding: 2rem;
  background: white;
  padding-bottom: 10rem; }

.content-image {
  display: flex;
  text-align: left;
  width: 20rem;
  word-wrap: break-word;
  justify-content: center;
  align-items: center; }

.welcome {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  flex-wrap: wrap;
  padding-top: 3rem; }

.welcome * {
  margin: 2rem;
  text-align: left;
  width: 20rem; }

.profiles {
  border-radius: 12rem;
  max-width: 10rem;
  height: 10rem;
  padding: 2rem 2rem; }

p.profile-title {
  margin-top: -1rem;
  max-width: 15rem;
  margin: 1rem auto;
  font-size: 1.5rem;
  color: #4d4d4d;
  text-shadow: -1px 1px 1px rgba(0, 0, 0, 0.22); }

p.profile-title-fancy {
  margin-top: -1rem;
  max-width: 15rem;
  margin: 1rem auto;
  color: white;
  text-shadow: -1px 1px 1px rgba(0, 0, 0, 0.22); }

.quote {
  font-size: 2rem;
  padding: 5rem 0; }

.bucket {
  display: flex;
  justify-content: space-around;
  align-content: center;
  border-radius: 0.5rem;
  background: #f8f8f8;
  margin: 5rem 0; }

.bucket-fancy {
  display: flex;
  justify-content: space-around;
  align-content: center;
  border-radius: 0.5rem;
  color: white;
  background-image: linear-gradient(200deg, #84fab0 0%, #8fd3f4 100%); }

.assessment {
  max-width: 2rem;
  padding: 0px;
  background: #00c5cc; }

.big-nav {
  font-size: 1.5rem;
  margin-bottom: 2rem; }

.dash-nav {
  display: flex;
  justify-content: flex-end; }

.box {
  margin-top: 2rem;
  max-width: 15rem;
  box-shadow: 0px 1px 3px rgba(128, 128, 128, 0.13);
  border-radius: 0.5rem; }

/*PLAYLIST
*********/
.playlist-container {
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  flex-wrap: wrap;
  background: #f8f8f8;
  padding: 1rem;
  border-radius: 2px; }

.playlist-card-contents {
  background: white;
  border-radius: 0.2rem;
  margin: 0.4rem;
  padding: 0.5rem;
  box-shadow: 0px 1px 3px rgba(128, 128, 128, 0.13); }

.playlist-card-contents:focus {
  outline: 0px; }

.email-form {
  align-self: center; }

.selected {
  border: 0.2px solid #33cfa4; }

.hidden {
  display: none; }

.song-title {
  padding: 0.5rem;
  color: #9d9d9d;
  text-align: left;
  max-width: 15rem;
  font-size: 12px; }

.delete {
  color: #d0d0d0; }

.delete:hover {
  color: #fa8072; }

/*SLANTDIVSTUFF
*********/
.diagonalbottom {
  background: white;
  height: 10rem;
  margin: -5rem 0 -1rem -1rem;
  width: 102vw;
  transform: skew(-2deg, 4deg);
  box-shadow: 0px 3px 0px #009eb8, 0px 8px 0px #00aea7, 0px 12px 0px rgba(51, 207, 164, 0.5), 0px 15px 0px rgba(51, 207, 164, 0.3), 0px 19px 0px rgba(51, 207, 164, 0.2); }

/* FORMS
*********/
form {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  text-align: center; }

form * {
  margin-bottom: 10px;
  border-radius: 4px;
  border: none;
  background: #f8f8f8;
  padding: 0.7rem; }

input::placeholder {
  color: #9d9d9d; }

form h5 {
  background: none; }

/*LINK
*********/
.link {
  color: #00c5cc; }

/* KEYFRAMES
*********/
.button-hover {
  cursor: pointer;
  display: inline-block;
  vertical-align: middle;
  transform: perspective(1px) translateZ(0);
  transition-duration: 0.3s;
  transition-property: transform;
  transition-timing-function: ease-out; }

.button-hover:hover, .button-hover:focus, .button-hover:active {
  transform: translateY(3px); }

/*DROPDOWN
*********/
.dropdown {
  display: inline-block; }

.dropdown__content {
  display: none;
  position: absolute; }

.dropdown--active .dropdown__content {
  display: flex;
  justify-content: flex-start;
  align-items: center; }

.dropdown-content {
  border-radius: 2px;
  padding-right: 4rem;
  box-shadow: 0 2px rgba(0, 0, 0, 0.05); }

button.button.dropdown-content {
  margin: 0.5rem 0; }

/*FOOTER
*********/
.footer {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  color: white;
  padding: 2rem;
  max-width: 10rem; }

.video-form-wrapper {
  padding: 1%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  width: 75vw;
  margin-left: 12.5vw;
  height: 15vw; }

.videos-list {
  width: 50%;
  min-height: 100%; }

.videos-form {
  width: 50%;
  min-height: 100%; }

.searchvideo {
  background-color: white; }

.searchvideo:hover {
  background-color: white;
  cursor: pointer; }

.searchinput:hover {
  cursor: pointer; }

/*MEDIA QUERIES*/
@media (max-width: 600px) {
  .logo {
    width: 3rem; }
  .banner {
    width: 200vw; }
  h1 {
    font-size: 2rem; }
  h2.body-bold {
    padding-top: 0.5rem;
    font-size: 1.5rem; }
  body {
    font-size: 1.5rem; }
  .content {
    padding-top: 1rem;
    margin-top: 1vmax; }
  .content-image {
    max-width: 15rem;
    padding: 1rem; } }

@media (max-width: 800px) and (min-width: 600px) {
  .banner {
    width: 250vw; }
  .logo {
    width: 3rem; }
  h1 {
    font-size: 2rem; }
  h2.body-bold {
    padding-top: 0.5rem;
    font-size: 1.5rem; }
  body {
    font-size: 1.5rem; }
  .content {
    padding-top: 1rem;
    margin-top: 1vmax; }
  .content-image {
    max-width: 12rem;
    margin: auto; } }

@media (max-width: 1200px) and (min-width: 800px) {
  .banner {
    width: 250vw; }
  body {
    font-size: 20px; }
  .logo {
    max-width: 3.5rem; }
  .button {
    font-size: 1rem; }
  .content-image {
    width: 10rem; } }

@media (min-width: 1200px) {
  .diagonalbottom {
    transform: skew(-2deg, 2deg); }
  .banner {
    width: 170vw; }
  body {
    font-size: 1rem; }
  .logo {
    width: 4.5rem; }
  .content {
    padding-top: 2rem;
    margin-top: 3rem; }
  .button {
    font-size: 1rem; }
  .content-image {
    width: 10rem; }
  h2.body-bold {
    margin-top: 0; } }

.grabbable {
  cursor: move;
  cursor: grab;
  text-align: center; }

.grabbable:active {
  cursor: move; }

div.spacearound {
  display: flex;
  justify-content: space-around;
  align-items: center; }

/*Timer
********/
.timer {
  width: 100px;
  height: 100px;
  background: #00c5cc;
  border-radius: 50px; }

.center {
  padding-top: 38px; }

#videoElement {
  width: 300px;
  background-color: #666; }
