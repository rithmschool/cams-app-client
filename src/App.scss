@import 'variables';

mainPage {
  font-family:$font-main;
  color: $dark-grey;
  background: $main-gradient;
  background-size: cover;
}

a {
  color: $main-blue;
  text-decoration: none;
}

.serif {
  font-family: $font-serif;
}

.white {
  color: white;
}

.bold-shadow {
  text-shadow: $heading-shadow;
  color: white;
}

.center {
  text-align: center;
  margin: auto;
}

.right {
  text-align: right;
}

.xs {
  font-size: .4rem;
}

.sm {
  font-size: .6rem;
}

.me {
  font-size: 1rem;
}

.lg {
  font-size: 1.5rem;
}

.xl {
  font-size: 2.5rem;
}

button {
  cursor: pointer;
}

/* NAV
*********/

.nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
  background: rgba(255, 255, 255, 0.9);
  border-bottom: 2px solid rgba(0, 0, 0, 0.1);
}

.logo {
  margin: 0.5rem;
  justify-content: flex-start;
  align-content: flex-start;
  border: 3px solid white;
  border-radius: 3px;
  box-shadow: $nav-shadow;
}

.button {
  background: white;
  border: 3px solid white;
  border-radius: 3px;
  padding: 0.7rem;
  box-shadow: $nav-shadow;
  color: $main-blue;
  margin: 0.5rem;
}

.button:hover {
  text-decoration: none;
}


/*BANNER
*********/

.banner {
  width: 120vw;
  z-index: -1;
  position: fixed;
  margin-top: -7rem;
}

.banner-text {
  padding: 0 1rem 0 1rem;
  color: white;
}

h1.banner-bold {
  color: white;
  max-width: 60vw;
  opacity: 0.8;
  font-size: 2.2em;
  font-style: italic;
}


/* CONTENT
*********/

.screens {
  padding-top: 5rem;
}

.content-image {
  display: flex;
  text-align: left;
  width: 20rem;
  word-wrap: break-word;
  justify-content: center;
  align-items: center;
}

.welcome {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  flex-wrap: wrap;
  padding-top: 3rem;
}

.welcome * {
  margin: 2rem;
  text-align: left;
  width: 20rem;
}
.profiles {
  border-radius: 12rem;
  max-width: 10rem;
  height: 10rem;
  padding: 2rem 2rem;
}
p.profile-title {
  margin-top: -1rem;
  max-width: 15rem;
  margin: 1rem auto;
  font-size: 1.5rem;
  color: $dark-grey;
  text-shadow: $title-shadow;
}
p.profile-title-fancy {
  margin-top: -1rem;
  max-width: 15rem;
  margin: 1rem auto;
  color: white;
  text-shadow: $title-shadow;
}

.quote {
  font-size: 2rem;
  padding: 5rem 0;
}
.bucket {
  display: flex;
  justify-content: space-around;
  align-content: center;
  border-radius: 0.5rem;
  background: $light-grey;
  margin:5rem 0;
  flex-wrap: wrap;
}
.bucket-fancy {
  display: flex;
  justify-content: space-around;
  align-content: center;
  border-radius: 0.5rem;
  color:white;
  background-image: $alt-gradient;
}
.assessment{
  max-width: 2rem;
  padding:0px;
  background:$main-blue;
}

.big-nav {
  font-size: 1.5rem;
  margin-bottom: 2rem;
}

.dash-nav {
  display: flex;
  justify-content: flex-end;
}

.box {
  margin-top: 2rem;
  max-width: 15rem;
  box-shadow: $nav-shadow;
  border-radius: 0.5rem;
}


/*PLAYLIST
*********/

.playlist-container {
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  flex-wrap: wrap;
  background: $light-grey;
  padding: 1rem;
  border-radius: 2px;
}

.playlist-card-contents {
  background: white;
  border-radius: 0.2rem;
  margin: 0.4rem;
  padding: 0.5rem;
  box-shadow: $nav-shadow;
}

.playlist-card-contents:focus {
  outline: 0px;
}

.email-form {
  align-self: center;
}

.selected {
  border: 0.2px solid $success;
}

.hidden {
  display: none;
}

.song-title {
  padding: 0.5rem;
  color:$bold-grey;
  text-align: left;
  max-width: 15rem;
  font-size: 12px;
}

.delete {
  color:rgb(208, 208, 208);
}

.delete:hover {
  color: $danger;
}


/*SLANTDIVSTUFF
*********/

.diagonalbottom {
  background: white;
  height: 10rem;
  margin: -5rem 0 -1rem -1rem;
  width: 110%;
  transform: skew(-2deg, 4deg);
  box-shadow: 0px 3px 0px rgb(0, 158, 184), 0px 8px 0px #00aea7, 0px 12px 0px rgba(51, 207, 164, 0.5), 0px 15px 0px rgba(51, 207, 164, 0.3), 0px 19px 0px rgba(51, 207, 164, 0.2);
}


/* FORMS
*********/

form {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  text-align: center;
}

form * {
  margin-bottom: 10px;
  border-radius: 4px;
  border: none;
  background: #f8f8f8;
  padding: 0.7rem;
}

input::placeholder {
  color:$bold-grey;
}

form h5 {
  background: none
}

form.email {}


/*LINK
*********/

.link {
  color: $main-blue;
}


/* KEYFRAMES
*********/

.button-hover {
  cursor: pointer;
  display: inline-block;
  vertical-align: middle;
  transform: perspective(1px) translateZ(0);
  transition-duration: 0.3s;
  transition-property: transform;
  transition-timing-function: ease-out;
}

.button-hover:hover, .button-hover:focus, .button-hover:active {
  transform: translateY(3px);
}


/*DROPDOWN
*********/

.dropdown {
  display: inline-block;
}

.dropdown__content {
  display: none;
  position: absolute;
}

.dropdown--active .dropdown__content {
  display: flex;
  justify-content: flex-start;
  align-items: center;
}

.dropdown-content {
  border-radius: 2px;
  padding-right: 4rem;
  box-shadow: 0 2px rgba(0, 0, 0, 0.05);
}

button.button.dropdown-content {
  margin: 0.5rem 0;
}


/*FOOTER
*********/

.footer {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  color: white;
  padding: 2rem;
  max-width: 10rem;
}

.videos-list {
  box-sizing: border-box;
  padding: 2%;
  padding-top: 0px;
  height: 25vw;
  width: 30vw;
  border: 0px solid;
  border-radius: 10px;
  background-color: #F1F1F1;
}

.createplaylistheader{
  height: 2.5vw;
}

.createplaylist{
  display: flex;
  justify-content: space-around;
}

.questions-form {
  box-sizing: border-box;
  padding: 2%;
  padding-top: 0px;
  height: 25vw;
  width: 30vw;
  border: 0px solid;
  border-radius: 10px;
  background-color: #F1F1F1;
}

.sortablelist{
  padding: 2%;
  padding-top: 0px;
  min-height: 25vw;
  width: 30vw;
  box-sizing: border-box;
  border: 0px solid;
  border-radius: 10px;
  background-color: #F1F1F1;
}

.grabbable{
  text-align: left;
}

.playlistlist{
  text-align: left;
}

.searchvideo{
  background-color: white;
}

.searchvideo:hover{
  background-color: white;
  cursor: pointer;
}

.searchinput{
  margin-bottom: 10px;
  border-radius: 4px;
  border: none;
  background: #f8f8f8;
  padding: 0.7rem;
}

.searchedvideos{
  overflow-y: scroll;
  height: 35%;
  width: 26vw;
  background-color: #f8f8f8;
  border-radius: 2%;
  color: black;
}

.searchitem{
  color: black;
}

.searchitem:hover{
  cursor: pointer
}


/*MEDIA QUERIES*/

@media (max-width: 600px) {
  .logo {
    width: 3rem;
  }
  .banner {
    width: 200vw;
  }
  h1 {
    font-size: 2rem;
  }
  h2.body-bold {
    padding-top: 0.5rem;
    font-size: 1.5rem;
  }
  body {
    font-size: 1.5rem;
  }
  .content {
    padding-top: 1rem;
    margin-top: 1vmax;
  }
  .content-image {
    max-width: 15rem;
    padding: 1rem;
  }
}

@media (max-width: 800px) and (min-width: 600px) {
  .banner {
    width: 250vw;
  }
  .logo {
    width: 3rem;
  }
  h1 {
    font-size: 2rem;
  }
  h2.body-bold {
    padding-top: 0.5rem;
    font-size: 1.5rem;
  }
  body {
    font-size: 1.5rem;
  }
  .content {
    padding-top: 1rem;
    margin-top: 1vmax;
  }
  .content-image {
    max-width: 12rem;
    margin: auto;
  }
}

@media (max-width: 1200px) and (min-width: 800px) {
  .banner {
    width: 250vw;
  }
  body {
    font-size: 20px;
  }
  .logo {
    max-width: 3.5rem;
  }
  .button {
    font-size: 1rem;
  }
  .content-image {
    width: 10rem;
  }
}

@media (min-width: 1200px) {
  .diagonalbottom{
    transform: skew(-2deg, 2deg);
  }
  .banner {
    width: 170vw;
  }
  body {
    font-size: 1rem;
  }
  .logo {
    width: 4.5rem;
  }
  .content {
    padding-top: 2rem;
    margin-top: 3rem;
  }
  .button {
    font-size: 1rem;
  }
  .content-image {
    width: 10rem;
  }
  h2.body-bold {
    margin-top: 0;
  }
}

.grabbable {
    cursor: move;
    cursor: grab;
    padding: 1%;
}


.grabbable:active {
    cursor: move;
}

div.spacearound {
  display: flex;
  justify-content: space-around;
  align-items: center;
}


/*Timer
********/

.timer {
  width: 100px;
  height: 100px;
  background: $main-blue;
  border-radius: 50px
}

.center {
  padding-top: 38px;
}

#videoElement {
    width: 500px;
    background-color: #666;
}

.content {
  margin-top: 1rem;
  text-align: center;
  padding: 2rem;
  background: white;
  padding-bottom: 10rem;
}