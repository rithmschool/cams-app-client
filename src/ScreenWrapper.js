import React, { PropTypes, Component } from "react";
import ScreenForm from "./ScreenForm";
import { BASE_URL, config, userID } from "./helpers.js";
import axios from "axios";
import getYouTubeID from "get-youtube-id";
import {
  SortableContainer,
  SortableElement,
  arrayMove
} from "react-sortable-hoc";

class ScreenWrapper extends Component {
  static contextTypes = {
    router: PropTypes.object
  };

  constructor(props) {
    super(props);
    this.state = {
      screenData: []
    };
    this.addVideo = this.addVideo.bind(this);
    this.addDone = this.addDone.bind(this);
    this.addQuestion = this.addQuestion.bind(this);
    this.addFile = this.addFile.bind(this);
  }

  addFile(target) {
    //var blob = new Blob([file], { type: "video/mp4" });
    var form_data = new FormData(target);
    debugger;
    return axios
      .post(
        `${BASE_URL}/api/files`,
        {
          data: form_data,
          contentType: false,
          processData: false,
          dataType: "json"
          //playlist_id: this.props.playlistID,
          //order: this.state.screenData.length + 1
        },
        config()
      )
      .then(response => {
        this.setState({
          screenData: this.state.screenData.concat([
            {
              title: response.data.title,
              entity_id: response.data.id,
              type: "video"
            }
          ])
        });
      });
  }

  addVideo(url) {
    let youtubeID = getYouTubeID(url);
    return axios
      .post(
        `${BASE_URL}/api/videos`,
        {
          url,
          youtube_id: youtubeID,
          playlist_id: this.props.playlistID,
          order: this.state.screenData.length + 1
        },
        config()
      )
      .then(response => {
        this.setState({
          screenData: this.state.screenData.concat([
            {
              title: response.data.title,
              entity_id: response.data.id,
              type: "video"
            }
          ])
        });
      });
  }

  addQuestion(question) {
    if (question) {
      return axios
        .post(
          `${BASE_URL}/api/questions`,
          {
            title: question,
            playlist_id: this.props.playlistID,
            order: this.state.screenData.length + 1
          },
          config()
        )
        .then(response => {
          this.setState({
            screenData: this.state.screenData.concat([
              {
                title: question,
                entity_id: response.data.id,
                type: "question"
              }
            ])
          });
        });
    }
  }

  addDone(url) {
    if (url) {
      this.addVideo(url).then(
        function(response) {
          this.context.router.history.push("/dashboard");
        }.bind(this)
      );
    }
    this.context.router.history.push("/dashboard");
  }

  onSortEnd = ({ oldIndex, newIndex }) => {
    this.setState({
      screenData: arrayMove(this.state.screenData, oldIndex, newIndex)
    });
  };

  componentWillMount() {
    if (this.props.editPlaylist) {
      axios
        .get(
          `${BASE_URL}/api/users/${userID()}/playlists/${this.props
            .playlistID}`,
          config()
        )
        .then(response => {
          this.setState({ screenData: response.data });
        });
    }
  }

  componentDidUpdate() {
    this.state.screenData.forEach((value, index) => {
      axios.patch(
        `${BASE_URL}/api/screens`,
        {
          entity_id: value.entity_id,
          playlist_id: this.props.playlistID,
          order: index + 1,
          type: value.type
        },
        config()
      );
    });
  }

  render() {
    const SortableVideo = SortableElement(({ value, key }) => (
      <li className="grabbable" key={key}>
        {value}
      </li>
    ));

    const SortableList = SortableContainer(({ screenData }) => {
      return (
        <ol>
          {screenData.map((value, index) => (
            <SortableVideo key={index + 1} index={index} value={value.title} />
          ))}
        </ol>
      );
    });

    return (
      <div>
        <h3 className="">Add Videos and Questions</h3>
        <SortableList
          screenData={this.state.screenData}
          onSortEnd={this.onSortEnd}
        />
        <ScreenForm
          addQuestion={this.addQuestion}
          addVideo={this.addVideo}
          addDone={this.addDone}
          addFile={this.addFile}
        />
      </div>
    );
  }
}

export default ScreenWrapper;
