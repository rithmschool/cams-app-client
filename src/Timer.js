import React from "react";
import PropTypes from "prop-types";

const Timer = ({ count }) => (
	<div>
		<div className="timer">
			<h2 className="center"> {count} </h2>
		</div>
	</div>
);

Timer.propTypes = {
	count: PropTypes.number.isRequired
};

export default Timer;
