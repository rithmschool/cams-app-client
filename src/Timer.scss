canvas {
  margin-left: auto;
  margin-right: auto;
  left: 0;
  right: 0;
}

.timer-wrapper {
  position: relative;
  width: 350px;
  height: 175px;
  margin-top: 20px;
}
