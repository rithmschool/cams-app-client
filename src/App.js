import React, {Component} from 'react';
import './index.css';
import './App.css';
import Edit from './Edit'
import banner from './banner.png';
import PlaylistWrapper from './PlaylistWrapper';
import Dashboard from './Dashboard';
import AssessmentsDashboard from './AssessmentsDashboard';
import LoginForm from './Login';
import Home from './Home';
import Nav from './Nav';
import {ensureCorrectUser} from './helpers'
import PatientWrapper from './PatientWrapper';
import {
  Route,
  Switch,
  Redirect
} from 'react-router-dom'
import './index.css';
import './App.css';


const PrivateRoute = ({ component, ...rest }) => (
  <Route {...rest} render={props => (
    localStorage.getItem('token') ? (
      React.createElement(component, props)
    ) : (
      <Redirect to={{
        pathname: '/login',
        state: { from: props.location }
      }}/>
    )
  )}/>
)

const EnsureCorrectUserRoute = ({ component, ...rest }) => (
  <Route {...rest} render={props => (
    ensureCorrectUser(props.match.params.userId) ? (
      React.createElement(component, props)
    ) : (
      <Redirect to={{
        pathname: '/login',
        state: { from: props.location }
      }}/>
    )
  )}/>
)

const EnsureLoggedOut = ({ component, ...rest }) => (
  <Route {...rest} render={props => (
    localStorage.getItem('token') ? (
      <Redirect to={{
        pathname: '/dashboard',
        state: { from: props.location }
      }}/>
    ) : (
      React.createElement(component, props)
    )
  )}/>
)

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {token: localStorage.getItem('token')}
  }

  componentWillReceiveProps() {
    this.setState({
      token: localStorage.getItem('token')
    })
  }

  render() {
    return (
      <div>
        <Nav isLoggedIn={!!this.state.token}/>
        <div>
          <img className="banner" src={banner} alt="patient and clinician smiling"/>
        </div>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/patient/home" component={PatientWrapper} />
          <EnsureLoggedOut exact path="/login" component={LoginForm} />
          <PrivateRoute path="/dashboard" component={Dashboard} />
          <PrivateRoute path="/assessments" component={AssessmentsDashboard}/>
          <PrivateRoute path="/playlists/new" component={PlaylistWrapper}/>
          <EnsureCorrectUserRoute path="/users/:userId/edit" component={Edit}/>
        </Switch>
        <div className="diagonalbottom"></div>
        <div className="footer">
          <p className="bold-shadow center">CAMS Corp</p>
          401 Parnassus Avenue
          San Francisco, CA 94143
        </div>
      </div>
    );
  }
}

export default App;
