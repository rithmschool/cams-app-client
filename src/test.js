import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import { Switch, MemoryRouter } from "react-router-dom";
import { shallow, mount } from "enzyme";
import ScreenWrapper from "./ScreenWrapper";
import renderer from "react-test-renderer";
import Timer from "./Timer";

class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}

global.localStorage = new LocalStorageMock();

test("shallow Home has correct components", () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find(Switch)).toHaveLength(1);
});

test("mount Home", () => {
  const wrapper = mount(
    <MemoryRouter initialEntries={["/"]} initialIndex={0}>
      <App />
    </MemoryRouter>
  );
  expect(wrapper.find(Switch)).toHaveLength(2);
});

test("snapshot of timer", () => {
  const timer = renderer.create(<Timer count={30} />).toJSON();
  expect(timer).toMatchSnapshot();
});
