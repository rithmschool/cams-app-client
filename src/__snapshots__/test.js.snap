exports[`test snapshot of timer 1`] = `
<div>
  <div
    className="timer">
    <h2
      className="center">
       
      30
       
    </h2>
  </div>
</div>
`;
